packer/packer.json

```diff
@@ -58,7 +58,7 @@
       ],
       "extra_arguments": [
         "--extra-vars",
-        "{\"common\": {\"upgrade_base\": \"true\" }, \"kubernetes\": {\"version\": \"{{user `kubernetes_version`}}\", \"cni_version\": \"{{user `kubernetes_cni_version`}}\"}}"
+        "common_upgrade_base=true kubernetes_version={{user `kubernetes_version`}} kubernetes_cni_version={{user `kubernetes_cni_version`}}"
```

Comment:
ah ha, all these changes lead to not having to have escaped json? Are there other reasons for the flattening of the vars?

Reply:
Not exactly, that is just an added benefit. The reason for flattening the vars is that if the var is a dict, then all values will need to be set when providing the dict. In this case, there were additional variables that were added for yum/apt repo information, and those would also need to be provided (or otherwise defaulted inline or through a filter plugin). The flattened vars allow for setting defaults individually on a per-var basis through normal ansible methods (defaults/main.yml).

This also aligns more closely with upstream Ansible best practices and convention.